File: algorithms/easy/number_of_distinct_averages.cpp
class Solution {
public:
    int distinctAverages(vector<int>& nums) {
      unordered_set<int> set;
      sort(nums.begin(), nums.end());
      for (int i = 0, j = nums.size() - 1; i < j; ++i, --j) {
        set.insert(nums[i] + nums[j]);
      }
      return set.size();
    }
};

File: algorithms/easy/number_of_distinct_averages.rs
use std::collections::HashSet;

impl Solution {
    pub fn distinct_averages(mut nums: Vec<i32>) -> i32 {
        nums.sort();
        let mut i = 0;
        let mut j = nums.len() - 1;
        let mut set = HashSet::new();
        while i < j {
            set.insert(nums[i] + nums[j]);
            i += 1;
            j -= 1;
        }
        set.len() as i32
    }
}
